File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if(root == null){
            return 0;
        }
        int level = 1;
        LinkedList<Pair> qt = new LinkedList<>();
        qt.offer(new Pair<>(root,1));
        while(!qt.isEmpty()){
            Pair<TreeNode,Integer> pair = qt.poll();
            TreeNode node = pair.getKey();
            level = pair.getValue();

            if(node.left != null){
                qt.offer(new Pair<>(node.left,level+1));
            }
            if(node.right != null){
                qt.offer(new Pair<>(node.right,level+1));
            }
        }
        return level;
        
    }
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        int depth = 0;
        if(root == nullptr){
            return depth;
        }
        queue<pair<TreeNode*,int>> qt;
        qt.push({root,1});
        while(!qt.empty()){
           auto [node,level] = qt.front();
           qt.pop();
           depth = max(depth,level);
           if(node->left != NULL){
            qt.push({node->left,level+1});
           }
           if(node->right != NULL){
            qt.push({node->right,level+1});
           }
        }
        return depth;
    }
};